### ISSUE TITLE
R package dependencies loading appropriately

### ISSUE THREAD

**Issue by mforde84** [spack/spack#10510]
Hi, I'm trying to install the CRAN package hdf5r. 

I'm receiving configuration errors indicating an incompatible system version of hdf5:

```
     38    Found hdf5 with version: 1.8.12
  >> 39    configure: error: The version of hdf5 installed on your system is not sufficient. Please ensure that at least version 1.8.13 is installed
```

I tried making my own package to include the hdf5 dependency:

Here's what the package.py looks like
```
from spack import *


class RHdf5r(AutotoolsPackage):
    url      = "file:///mnt/beegfs/build/mirror/r-hdf5r/r-hdf5r-1.0.1.tar.gz"
    version('1.0.1', sha256='32f35dc26809a7e9b9335e0dc24369cf87f4c6deb430b2d9d019450a10164240')

    depends_on('hdf5', type=('build', 'run'))
    depends_on('r', type=('build', 'run'))
```

And here's the installation attempt specifying R, HDF5 hashes, plus the appropriate C AND LD flags:
```
16:38:07-root@gosset:~$ spack find --paths | grep -E 'hdf5|r@3.5.2'
    hdf5@1.8.13                   /mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/hdf5-1.8.13-ymstonfndqgej3ribn7c6jtckpolwdgp
    r@3.5.2                       /mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/r-3.5.2-6sd4xtdyjin77zkwiasoa3iea4ukpttq
17:27:27-root@gosset:~$ h5dump --version
h5dump: Version 1.8.13
17:28:37-root@gosset:~$ spack install r-hdf5r %gcc@7.4.0 ldflags="-L/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/hdf5-1.8.13-ymstonfndqgej3ribn7c6jtckpolwdgp/lib" cflags="-I/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/hdf5-1.8.13-ymstonfndqgej3ribn7c6jtckpolwdgp/include" ^/6sd4xtdyjin77zkwiasoa3iea4ukpttq ^/ymstonfndqgej3ribn7c6jtckpolwdgp
...
==> Building r-hdf5r [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/mnt/beegfs/spackroot/var/spack/stage/r-hdf5r-1.0.1-siozbz7bebvhdd4pydkpq5wy5eyaf3s2/hdf5r/configure' '--prefix=/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/r-hdf5r-1.0.1-siozbz7bebvhdd4pydkpq5wy5eyaf3s2'

1 error found in build log:
     33    checking for H5Fcreate in -lhdf5... yes
     34    checking for hdf5_hl.h... no
     35    checking for H5LTpath_valid in -lhdf5_hl... yes
     36    checking for main in -lhdf5_hl... yes
     37    checking for matching HDF5 Fortran wrapper... /bin/h5fc
     38    Found hdf5 with version: 1.8.12
  >> 39    configure: error: The version of hdf5 installed on your system is not sufficient. Please ensure that at least version 1.8.13 is installed
```
Yet it doesn't appear the compilation is using the appropriate pkg configuration and is still referencing the older version of hdf5. 

Trying to install the package with R CMD INSTALL fails in the same manner:
```
17:40:35-root@gosset:/mnt/beegfs/build/mirror/r-hdf5r$ R CMD INSTALL --configure-args="LDFLAGS=-L/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/hdf5-1.8.13-ymstonfndqgej3ribn7c6jtckpolwdgp/lib CFLAGS=-I/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/hdf5-1.8.13-ymstonfndqgej3ribn7c6jtckpolwdgp/include" r-hdf5r-1.0.1.tar.gz
...
Found hdf5 with version: 1.8.12
configure: error: The version of hdf5 installed on your system is not sufficient. Please ensure that at least version 1.8.13 is installed
ERROR: configuration failed for package ‘hdf5r’
* removing ‘/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/r-3.5.2-6sd4xtdyjin77zkwiasoa3iea4ukpttq/rlib/R/library/hdf5r’
```
Any suggestions?

**Comment by mforde84** [comment]
I think I have it. After consulting the configuration file, the --with-hdf5 flag was asking for the path of h5pcc, and the following worked correctly:

```
./configure --with-hdf5=/mnt/beegfs/spackroot/opt/spack/linux-centos7-x86_64/gcc-7.4.0/hdf5-1.10.4-idpd5moo5pr2c4dfq23wmatr4zmn2zwu/bin/h5pcc
```

So chop it up to a piss poor configuration file.  Which question though, I know that I can spec this using python through:

spec['hdf5'].{some attribute}

Is there documentation for a list of available attributes, or that I can query in an interactive session of python to test whether my python is sane. I'm relatively new to py, so I'm trying to fill in the gaps as best I can.

M

**Comment by adamjstewart** [comment]
Closed by #12019, Spack now has a builtin package for `r-hdf5r`.
